Document how to configure XContent parsing constraints (#4154)
opensearch-project/documentation-website: _install-and-configure/install-opensearch/index.md

Do not declare the same JVM options in multiple locations because it can result in unexpected behavior or a failure of the OpenSearch service to start. If you declare JVM options using an environment variable, such as `OPENSEARCH_JAVA_OPTS=-Xms3g -Xmx3g`, then you should comment out any references to that JVM option in `config/jvm.options`. Conversely, if you define JVM options in `config/jvm.options`, then you should not define those JVM options using environment variables.
{: .note}

## Important system properties

> The OpenSearch has a number of system properties that you can specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS` using `-D` command line argument notation.

Reviewer: OK sure. Then how about this
```suggestion
The OpenSearch has a number of system properties that you can specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS` using `-D` command line argument notation, for example:
```

> The OpenSearch has a number of system properties that you could specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS`, using `-D` command line arguments notation.

Reviewer: ```suggestion
The OpenSearch has a number of system properties that you can specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS` using `-D` command line argument notation.
```

> - `opensearch.xcontent.string.length.max=<value>`

Reviewer: Could we add an introductory sentence for this snippet? How about this:
````suggestion
For example, the following setting sets the `opensearch.xcontent.string.length.max` to `<value>`:

```plaintext
- opensearch.xcontent.string.length.max=<value>
```
````

Author: My intention here was to keep it as a lit, similarly to the previous section (not changed here) with system parameters.

Reviewer: Understood. See my suggestion above.

>   By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your clusters from pontential DDoS or memory issues, you could set the system property `opensearch.xcontent.string.length.max` to the reasonable boundary (the maximum is `2147483647`), for example: `-Dopensearch.xcontent.string.length.max=5000000`.

Reviewer: ```suggestion
By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your cluster from potential distributed denial-of-service (DDoS) or memory issues, you can set the `opensearch.xcontent.string.length.max` system property to a reasonable limit (the maximum is 2,147,483,647), for example `-Dopensearch.xcontent.string.length.max=5000000`.
```

> By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your cluster from potential distributed denial-of-service (DDoS) or memory issues, you can set the `opensearch.xcontent.string.length.max` system property to a reasonable limit (the maximum is 2,147,483,647), for example `-Dopensearch.xcontent.string.length.max=5000000`.

Reviewer 2: ```suggestion
By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your cluster from potential distributed denial-of-service (DDoS) or memory issues, you can set the `opensearch.xcontent.string.length.max` system property to a reasonable limit (the maximum is 2,147,483,647), for example, `-Dopensearch.xcontent.string.length.max=5000000`.
```
